fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreSplitRecordsTest.java

```diff
@@ -800,22 +869,52 @@ private void checkForConflicts(int formatVersion, boolean splitLongRecords, @Non
-                    store2.saveRecord(TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(1415L).build());
-                });
+        SplitRecordsTestConfig.allConfigs().forEach(config -> {
+            this.testConfig = config;
+            checkForConflicts(formatVersion, splitLongRecords,
+                    store1 -> store1.saveRecord(TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(1066L).build()),
+                    store2 -> {
+                        store2.addRecordReadConflict(Tuple.from(1066L));
+                        store2.saveRecord(TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(1415L).build());
+                    });
+        });
     }
 
     @ParameterizedTest(name = "recordWriteConflict [formatVersion = {0}, splitLongRecords = {1}]")
     @MethodSource("formatVersionAndSplitArgs")
     public void recordWriteConflict(int formatVersion, boolean splitLongRecords) {
-        checkForConflicts(formatVersion, splitLongRecords,
-                store1 -> store1.addRecordWriteConflict(Tuple.from(1066L)),
-                store2 -> {
-                    store2.loadRecord(Tuple.from(1066L));
-                    store2.saveRecord(TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(1415L).build());
-                });
+        SplitRecordsTestConfig.allConfigs().forEach(config -> {
+            this.testConfig = config;
+            checkForConflicts(formatVersion, splitLongRecords,
+                    store1 -> store1.addRecordWriteConflict(Tuple.from(1066L)),
+                    store2 -> {
+                        store2.loadRecord(Tuple.from(1066L));
+                        store2.saveRecord(TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(1415L).build());
+                    });
+        });
+    }
+
+    @ParameterizedTest(name = "recordDeleteConflict [formatVersion = {0}, splitLongRecords = {1}]")
+    @MethodSource("formatVersionAndSplitArgs")
+    public void recordDeleteConflict(int formatVersion, boolean splitLongRecords) {
+        SplitRecordsTestConfig.allConfigs().forEach(config -> {
+            this.testConfig = config;
+            try (FDBRecordContext context = openContext()) {
+                final RecordMetaDataHook hook = metaData -> metaData.setSplitLongRecords(splitLongRecords);
+                FDBRecordStore recordStore = getStoreBuilder(context, simpleMetaData(hook))
+                        .setFormatVersion(formatVersion)
+                        .createOrOpen();
+                // Insert a record here so that the delete logic (1) sees the record is there and then
+                // (2) issues a real delete
+                recordStore.saveRecord(TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(1066L).build());
+                commit(context);
+            }
+            checkForConflicts(formatVersion, splitLongRecords,
+                    store1 -> store1.deleteRecord(Tuple.from(1066L)),
```

Comment:
Would there be benefit to adding another version that does a save, and then gets a conflict on the delete?

Reply:
Sure, added